integrations/messenger/src/index.ts

```diff
@@ -268,30 +270,47 @@ export function getRecipientId(conversation: SendMessageProps['conversation']):
   return recipientId
 }
 
-async function handleMessage(message: MessengerMessage, client: bp.Client) {
-  if (message.message) {
-    if (message.message.text) {
-      const { conversation } = await client.getOrCreateConversation({
-        channel: 'channel',
-        tags: {
-          [idTag]: message.sender.id,
-        },
-      })
-
-      const { user } = await client.getOrCreateUser({
-        tags: {
-          [idTag]: message.sender.id,
-        },
-      })
-
-      await client.createMessage({
-        tags: { [idTag]: message.message.mid },
-        type: 'text',
-        userId: user.id,
-        conversationId: conversation.id,
-        payload: { text: message.message.text },
-      })
+async function handleMessage(
+  message: MessengerMessage,
+  {
+    client,
+    ctx,
+    logger,
+  }: { client: bp.Client; ctx: IntegrationContext<bp.configuration.Configuration>; logger: IntegrationLogger }
+) {
+  if (message?.message?.text) {
+    const { conversation } = await client.getOrCreateConversation({
+      channel: 'channel',
+      tags: {
+        [idTag]: message.sender.id,
+      },
+    })
+
+    const { user } = await client.getOrCreateUser({
+      tags: {
+        [idTag]: message.sender.id,
+      },
+    })
+
+    // TODO: do this for profile_pic as well, as of 13 NOV 2023 the url "https://platform-lookaside.fbsbx.com/platform/profilepic?eai=<eai>&psid=<psid>&width=<width>&ext=<ext>&hash=<hash>" is not working
+    if (!user.name) {
+      try {
+        const messengerClient = getMessengerClient(ctx.configuration)
+        const profile = await messengerClient.getUserProfile(message.sender.id)
+
+        await client.updateUser({ ...user, name: profile.name })
+      } catch (error) {
+        logger.forBot().error('Error while fetching user profile', error)
```

Comment:
For clarity, to avoid confusing it with an issue with fetching the user from our Bridge API.

```suggestion
        logger.forBot().error('Error while fetching user profile from Messenger:', error)
```